soccer/gameplay/tactics/advance_zone_midfielder.py

```diff
@@ -0,0 +1,174 @@
+
+        self.dribble_state = True
+
+        self.kick = False
+
+        for s in AdvanceZoneMidfielder.State:
+            self.add_state(s, behavior.Behavior.State.running)
+
+        self.names = ['left','right']
+        #intially should be passSet method
+        self.add_transition(behavior.Behavior.State.start,
+                            AdvanceZoneMidfielder.State.passSet, lambda: True,
+                            "Immediately")
+        self.add_transition(AdvanceZoneMidfielder.State.passSet,
+                            AdvanceZoneMidfielder.State.hold, lambda: False,
+                            "When a kick play begins")
+        self.add_transition(AdvanceZoneMidfielder.State.hold,
+                            AdvanceZoneMidfielder.State.passSet, lambda: True,
+                            "When not in a kick play")
+
+    def on_enter_passSet(self):
+        print("Entering PassSet");
+
+    def execute_passSet(self):
+        #gets the best position to travel to for ball reception
+        best_Pos_From_Ball = self.passing_point
+        """if (self.dribble_state):
+            points = [robocup.Point(0, 0), robocup.Point(0, 0)]
+            self.optimal = 1
+            # sets either the left or the right side of the ball to the best ball position
+            # and then notes which one is the other
+            # left side of the ball
+            points[self.optimal] = best_Pos_From_Ball
+            self.other = 0
+        else :
+            points = [robocup.Point(0, 0), robocup.Point(0, 0)]
+            self.optimal = 1
+            # sets either the left or the right side of the ball to the best ball position
+            # and then notes which one is the other
+            # left side of the ball
+            points[self.optimal] = best_Pos_From_Ball
+            self.other = 0
+        """
+        points = [robocup.Point(0, 0), robocup.Point(0, 0)]
+        points[0] = best_Pos_From_Ball;
+
+        # sets the second point
+        pointer, value2 = evaluation.passing_positioning.eval_best_receive_point(
+                best_Pos_From_Ball,
+                main.our_robots(), AdvanceZoneMidfielder.FIELD_POS_WEIGHTS,
+                AdvanceZoneMidfielder.NELDER_MEAD_ARGS,
+                AdvanceZoneMidfielder.PASSING_WEIGHTS)
+        
+        points[1] = pointer
+        # moves the robots
+        for i in range(2):
+            if (self.moves[i] is None):
+                self.moves[i] = skills.move.Move(points[i])
+                self.add_subbehavior(
+                    self.moves[i], self.names[i], required=False, priority= self.priorities[i])
+            else:
+                self.moves[i].pos = points[i]
+
+    def on_enter_hold(self):
+        print("kicking")
+
+    def execute_hold(self):
+
+        y_temp_hold = 0.8 * self.passing_point.y
+
+        x_temp_hold = constants.Field.Width / 3
+
+        self.hold_point = [robocup.Point(x_temp_hold,y_temp_hold), robocup.Point(x_temp_hold,y_temp_hold)]
```

Comment:
These are the same definitions that are in the constructor. I would rather remove these and leave the ones in the constructor if they are just going to be static variables.